### ISSUE TITLE
spikegen.latency returns float32 for float64 inputs

### ISSUE THREAD

**Issue by GraceDyer** [jeshraghian/snntorch#440]
**Issue type**

- [x] Bug Report
- [ ] Feature Request
- [ ] Help wanted
- [ ] Other

**snnTorch version**

`1.0.0`

**Description**

I found a reproducible dtype bug in the public encoding helper `snntorch.spikegen.latency`.

When the input tensor uses `torch.float64`, the generated spike tensor is returned as `torch.float32`. This occurs on the default path and when using the `normalize`, `linear`, or `interpolate` options.

As a result, latency encoding silently changes the caller's floating-point dtype. This can cause precision loss or a dtype mismatch when the encoded spikes are passed to a model converted with `.double()`.

From source inspection, `spikegen.py` defines a module-level `dtype = torch.float`. The latency path creates its output tensor with `torch.zeros(..., dtype=dtype)`, so the generated spikes use the fixed `float32` dtype rather than the dtype of the input tensor.

I reproduced this with both snnTorch `1.0.0` and the current master branch in fresh CPU-only Docker containers using Debian GNU/Linux 12, Python 3.11.16, and PyTorch 2.8.0+cpu.

**Minimal code to reproduce the error/bug**

```python
import torch
from snntorch import spikegen

x = torch.tensor([0.2, 0.8], dtype=torch.float64)

configurations = {
    "default": {},
    "normalize": {"normalize": True},
    "linear+normalize": {"linear": True, "normalize": True},
    "interpolate+normalize": {"interpolate": True, "normalize": True},
}

print("input_dtype =", x.dtype)

for name, kwargs in configurations.items():
    output = spikegen.latency(x, num_steps=5, **kwargs)
    print(name, "output_dtype =", output.dtype)
```

**What is expected**

`spikegen.latency` should preserve the floating-point dtype of the input tensor.

For a `torch.float64` input, the generated spike tensors should also use `torch.float64`.

**What actually happens**

All tested latency-encoding paths return `torch.float32` tensors for a `torch.float64` input.

Example output from the Docker reproduction:

```text
input_dtype = torch.float64
default output_dtype = torch.float32
normalize output_dtype = torch.float32
linear+normalize output_dtype = torch.float32
interpolate+normalize output_dtype = torch.float32
```

**Comment by ixfd64** [comment]
Should be fixed in #445. Let us know if you still see issues.
